app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ReviewRegistrationController.scala

```diff
@@ -159,48 +160,44 @@ class ReviewRegistrationController @Inject() (
     registration.organisationDetails.organisationType.flatMap {
       case OrgType.SOLE_TRADER =>
         registration.organisationDetails.soleTraderDetails.flatMap(
-          details => details.registration.registeredBusinessPartnerId
+          soleTraderDetails => soleTraderDetails.registration.registeredBusinessPartnerId
         )
       case OrgType.PARTNERSHIP =>
         registration.organisationDetails.partnershipDetails.flatMap(
           partnershipDetails =>
             partnershipDetails.partnershipType match {
               case GENERAL_PARTNERSHIP =>
-                partnershipDetails.generalPartnershipDetails.fold(
-                  throw new IllegalStateException("Missing general partnership details")
-                )(_.registration.registeredBusinessPartnerId)
+                partnershipDetails.generalPartnershipDetails.flatMap(
+                  generalPartnershipDetails =>
+                    generalPartnershipDetails.registration.registeredBusinessPartnerId
+                )
               case SCOTTISH_PARTNERSHIP =>
-                partnershipDetails.scottishPartnershipDetails.fold(
-                  throw new IllegalStateException("Missing scottish partnership details")
-                )(_.registration.registeredBusinessPartnerId)
+                partnershipDetails.scottishPartnershipDetails.flatMap(
+                  scottishPartnershipDetails =>
+                    scottishPartnershipDetails.registration.registeredBusinessPartnerId
+                )
               case _ => throw new IllegalStateException("Illegal partnership type")
             }
         )
       case _ =>
         registration.organisationDetails.incorporationDetails.flatMap(
           details => details.registration.registeredBusinessPartnerId
         )
-    }.getOrElse(throw new Exception("Safe Id is required for a Subscription create"))
+    }.getOrElse(throw new IllegalStateException("Safe Id is required for a Subscription create"))
 
   private def getSoleTraderDetails()(implicit
     request: JourneyRequest[AnyContent]
   ): Future[SoleTraderIncorporationDetails] =
-    request.registration.organisationDetails.soleTraderDetails.fold(
-      throw new Exception("Unable to fetch sole trader details from cache")
-    )(details => Future.successful(details))
+    Future.successful(request.registration.organisationDetails.soleTraderDetails.get)
```

Comment:
Was going to comment on the use of "get".
But looking at the usage of this method we are creating a Future from an Option just so that we can immediately un-wrap it (the Future) and then pass it to the view as an Option.

Could do with some refactoring?

Reply:
Yeah, I'll see to it.